x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/oidc/OpenIdConnectAuthenticateResponse.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.security.action.oidc;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.io.IOException;
+
+public class OpenIdConnectAuthenticateResponse extends ActionResponse {
+    private String principal;
+    private String accessTokenString;
```

Comment:
We have traditionally not used `SecureString` for the token values that we generate in the `TokenService`.  I think there are fundamental differences in the threat model of passwords (for which we usually use `SecureString`) and token values. 

We're dealing with an attacker that has read access to the JVMs memory contents and can dump the heap. For passwords, it might make sense to protect against that threat, as the password is user specific and can be used to persist access or compromise the user in other systems. For the token values now, it feels like it's not worth doing. For the case for access tokens especially since they are short lived. 

I wouldn't mind discussing this further and/or use secure string, but we could do this in a follow up and cover all cases of tokens usage, and not just the response object of this API. WDYT ?